tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb

```diff
@@ -492,36 +510,80 @@ def puppet_resource_cmd(af)
   get_namespace_cmd(agent, cmd, options)
 end
 
-# Search pattern for show run config testing
-def af_pattern(tests, id, af)
-  asn, vrf, nbr, afi, safi = af.values
-  if tests[id][:ensure] == :present
-    if vrf[/default/]
-      [/router bgp #{asn}/, /neighbor #{nbr}/,
-       /address-family #{afi} #{safi}/]
+def get_dependency_manifest(platform, af, remote)
+  extra_config = ''
+  if platform == 'ios_xr'
+    # XR requires the following before a vrf AF can be configured:
+    #   1. a global router_id
+    #   2. a global address family
+    #   3. route_distinguisher configured on the vrf
+    #   4. remote-as is required for neightbor
+    remote = 2 if remote.nil?
+    if af[:vrf] == 'default'
+      extra_config = "
+      cisco_bgp { '#{af[:asn]} #{af[:vrf]}':
```

Comment:
Looking at this some more, it seems like you are doing more "dependency setup" than is required.

In general, I don't think you have to create any of the "parent" instances (bgp, bgp_af) unless there are non-default values you need to set (like the bgp router_id when configuring an af under a vrf).

I don't know the details behind each of these, so I could certainly be wrong, but it seems like the following would be enough:

On XR:
- configure a bgp instance with a router_id (not sure if this is needed for both default & non-default vrf cases)
- configure a "parent" vpnv4 af
- configure any route-policies you need
- when vrf != default, configure a bgp vrf instance with a route_distinguisher set
- not sure about the neighbor with a remote_as, it looks like it is needed based on your comments/code

It looks like the neighbor is required for nexus, as well, so maybe that should just be moved out of the "if ios_xr" block?